universal-application-tool-0.0.1/app/services/Path.java

```diff
@@ -127,21 +127,27 @@ public Path withoutArrayReference() {
   /**
    * Return the index of the array element this path is referencing.
    *
-   * <p>For paths to non-array elements, {@code IllegalStateException is thrown}.
+   * <p>For paths to non repeated entity collections, or paths to an entire repeated entity
+   * collection without specifying an index, {@code IllegalStateException} is thrown.
```

Comment:
Suggestion (blocking): please clarify that this method will return the *last* array index in the path.

Reply:
done